### ISSUE TITLE
cannot cast to a Callable

### ISSUE THREAD

**Issue by dbarbier** [microsoft/pyright#271]
Pyright seems unable to `cast()` to a Callable.  Here is a contrived example:

```
from typing import Callable, cast

FUNC = Callable[[int], int]
def foo(i: int) -> int:
    return 42

bar = cast(FUNC, foo) # pyright error here
print(bar(9))
print(bar('9'))
```
VS Code reports an error at the line defining `bar`:
```
No overloads for 'cast(FUNC, foo)' match parameters
  Argument types: ((p0: int) -> int, (i: int) -> int)
```

**Expected behavior**
IMHO pyright should happily cast `bar` to `FUNC`, and report an error at the last line when `bar` argument is not an `int`.

**VS Code extension or command-line**
VS Code extension 1.0.67

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by dbarbier** [comment]
Wow, thanks for your quick fix; I built 8b6c825 and confirm that it fixes my real use case.

**Comment by erictraut** [comment]
This is now fixed in 1.0.68, which I just published.
